### ISSUE TITLE
[tcgc] have `@clientInitialization` decorator take in optional `access` input

### ISSUE THREAD

**Issue by iscai-msft** [Azure/typespec-azure#1696]
### Clear and concise description of the problem

Currently, the `@clientInitialization` decorator solely takes in a required model input. However, as we are trying to move away from the `@operationGroup` decorator, we want to use the `@clientInitialization` decorator to also specify whether a client is publicly or internally instantiable. We are currently using client hierarchy and the `@operationGroup` decorator to determine whether the initialization on a client is public or not. Instead, we should almost do the opposite. Get rid of the `@operationGroup` decorator, based off of client hierarchy, we can default over whether a client is publicly or internally instantiable, but allow users to specify the access level of initialization.

// PROPOSED CHANGES TO @CLIENTINITIALIZATION DECORATOR

// Client with public initialization -> maps to current def of client
// Client with internal initialization -> maps to current def of operation group

// current design for @clientInitialization(client, model, scope?)
// proposed designs for 

// 1. @clientInitialization(client, model?, scope?)
// 2. @clientInitialization(client, access?, scope?)
// 3. @clientInitialization(client, model?, access?, scope?)

// @clientInitialization(client, {model?: Model, access?: AccessFlags}, scope?: string)

// @@operationGroup(MyService) -> @@clientInitialization(MyService, {access: "internal"})
// @@clientInitialization(MyService, {model: MyClientInitialization})

### Checklist

- [x] Follow our [Code of Conduct](https://github.com/azure/typespec-azure/blob/main/CODE_OF_CONDUCT.md)
- [x] Check that this issue is about the Azure libraries for typespec. For feature request in the typespec language or core libraries file it in the [TypeSpec repo](https://github.com/Microsoft/TypeSpec/issues/new/choose)
- [x] Read the [docs](https://azure.github.io/typespec-azure/).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by weidongxu-microsoft** [comment]
What would you write for an Interface as subclient (with clientInitialization access=internal, as usual) without `@operationGroup`?

**Comment by weidongxu-microsoft** [comment]
Also, I assume there is a complementary concept in SdkClientAccessor

```
export interface SdkClientAccessor<TServiceOperation extends SdkServiceOperation>
  extends SdkMethodBase {
  kind: "clientaccessor";
  response: SdkClientType<TServiceOperation>;
}
```

For a subclient accessed via `client.getSubClient`, I'd assume `SdkInitializationType.access=internal` and `SdkClientAccessor.access=public`?

**Comment by iscai-msft** [comment]
> What would you write for an Interface as subclient (with clientInitialization access=internal, as usual) without `@operationGroup`? Or we need the feature of nested Interface?

@tadelesh can you correct me if I'm wrong, but in the case of no `@client` decorator, if you have a namespace with an internal interface, we treat the internal interface automatically as a subclient with `internal` initialization. So you wouldn't have to do anything in this scenario

**Comment by weidongxu-microsoft** [comment]
> > What would you write for an Interface as subclient (with clientInitialization access=internal, as usual) without `@operationGroup`? Or we need the feature of nested Interface?
> 
> [@tadelesh](https://github.com/tadelesh) can you correct me if I'm wrong, but in the case of no `@client` decorator, if you have a namespace with an internal interface, we treat the internal interface automatically as a subclient with `internal` initialization. So you wouldn't have to do anything in this scenario

Yes, but in such case we don't need to write `@operationGroup` either. Namespace and interface implicitly give the client/subclient structure.

But when we need to write `@client` in client.tsp, things changed. Previously we need to write a `@operationGroup` to make an interface a subclient. What would we do now? (or we plan to remove the `@client` too, and let namespace/interface in client.tsp service the client/subclient structure?)

**Comment by tadelesh** [comment]
@weidongxu-microsoft 's point is right, the current client hierarchy has two situations:
1. if there is no `@client` and `@operationGroup` defined, then we will treat any namespaces with `@service` to be the root client, and any namespaces and interfaces under that namespace (with `@service`) will be the sub-clients.
2. if there is `@client` or `@operationGroup` defined, then the client hierarchy will only follow these two decorators, `@client` define the clients and `@operationGroup` defined the sub-clients.
if we remove the `@operationGroup`, we need to get consensus about how to detect the sub-clients, two proposals:
1. always follow namespaces and interfaces from namespace, add `@ignoreClient` something like that to ignore the special case
2. reuse `@client` to decorate the sub-clients, and follow current rule, if use `@client`, then define all the clients with `@client`, this is somehow replace `@operationGroup` with `@client`
